src/formats/tomboy/core.ts

```diff
@@ -0,0 +1,605 @@
+	 * Track the most recent TODO heading for context-aware TODO mode
+	 */
+	private currentTodoHeading: { level: number } | null = null;
+
+	/**
+	 * Mode for keeping titles in markdown
+	 */
+	private keepTitleMode: KeepTitleMode = 'automatic';
+
+	/**
+	 * Enable or disable TODO list processing
+	 */
+	setTodoEnabled(enabled: boolean): void {
+		this.todoEnabled = enabled;
+	}
+
+	/**
+	 * Set the mode for keeping titles in markdown
+	 */
+	setKeepTitleMode(mode: KeepTitleMode): void {
+		this.keepTitleMode = mode;
+	}
+
+	/**
+	 * Check if all sections of a list item have strikethrough formatting (simplified requirement)
+	 */
+	private isFullyStrikethrough(sections: Array<ContentSection>): boolean {
+		return sections.every(section => section.xmlPath.includes('strikethrough'));
+	}
+
+	/**
+	 * Check if a note title contains TODO keywords
+	 */
+	private isTodoTitle(title: string): boolean {
+		const todoKeywords = ['TODO', 'Todo', 'To Do', 'todo', 'to do', 'ToDo'];
+		return todoKeywords.some(keyword => title.includes(keyword));
+	}
+
+	/**
+	 * Update TODO context based on heading level and text
+	 */
+	private updateTodoContext(headingLevel: number, headingText: string): void {
+		// Only update if this heading is at same level or higher than current TODO heading
+		if (!this.currentTodoHeading || headingLevel <= this.currentTodoHeading.level) {
+			const isTodo = this.isTodoTitle(headingText);
+			this.currentTodoHeading = isTodo
+					? { level: headingLevel }
+					: null;
+		}
+	}
+
+	private isTodoMode() : boolean {
+		return this.todoEnabled && this.currentTodoHeading !== null;
+	}
+
+	/**
+	 * Parse a Tomboy date string into a Date object
+	 * Tomboy format: 2025-09-10T21:11:41,964692Z
+	 * Converts comma to period for standard parsing
+	 */
+	private parseTomboyDate(dateStr: string): Date {
+		// Convert Tomboy date format (2025-09-10T21:11:41,964692Z) to standard format
+		const standardDateStr = dateStr.replace(',', '.');
+		return new Date(standardDateStr);
+	}
+
+	/**
+	 * Parse Tomboy XML content into structured format
+	 */
+	parseTomboyXML(xmlContent: string): TomboyNote {
+		const parser = new DOMParser();
+		const doc = parser.parseFromString(xmlContent, 'text/xml');
+
+		const title = doc.querySelector('title')?.textContent || 'Untitled';
+		const textElement = doc.querySelector('text');
+
+		let content: Array<ContentLine> = [];
+		if (textElement) {
+			content = this.parseContentStructure(textElement);
+		}
+
+		const tagsElement = doc.querySelector('tags');
+		const tags = tagsElement?.textContent ? tagsElement.textContent.split(',').map(tag => tag.trim()) : [];
+
+		// Parse date elements
+		const createDateElement = doc.querySelector('create-date');
+		const lastChangeDateElement = doc.querySelector('last-change-date');
+		
+		let createDate: Date | undefined;
+		let lastChangeDate: Date | undefined;
+
+		if (createDateElement?.textContent) {
+			createDate = this.parseTomboyDate(createDateElement.textContent);
```

Comment:
If `parseTomboyDate` returned null if it was passed null, this section of code could be simplified nicely.